src/Components/Browser.JS/src/Boot.Server.ts

```diff
@@ -95,7 +95,16 @@ async function boot(userOptions?: Partial<BlazorOptions>): Promise<void> {
     circuits.push(circuit);
   }
 
-  await reconnectTask;
+  if (!await reconnectTask) {
```

Comment:
We should return something from the circuit to specifically signal that the circuit is gone and then do different things based on whether we were able to ever connect to the circuit.

* If we prerendered, closed and reopened, then it's fair game to reload.
* If not, we should simply display different UI explaining what happened and offering the user to reload the app.
  * Its a very bad user experience if you go through a tunnel or close your laptop for 1 minute and then suddenly your 
    app gets refreshed without you knowing why and potentially loosing data/etc.
  * Apps should be able to customize what happens in this case, for example to save form fields in local storage and so on before refreshing.

Reply:
> Apps should be able to customize what happens in this case, for example to save form fields in local storage and so on before refreshing.

That sounds great, but I'd be OK with considering it out of scope for this PR. It sounds like a bigger feature change. This PR doesn't regress the existing situation, since there isn't yet a way of doing that anyway.

Reply:
As long as we can detect in a deterministic way that the circuit is gone (the server says so) we shouldn't reload automatically. Other than that, although I would like to see app developers having a choice about when the reload happens, I'm ok with simply reloading for now (only when we know for sure the circuit is gone).